spec/models/reservation_spec.rb

```diff
@@ -577,5 +577,33 @@
     end
   end
 
+  context 'check in check out validations' do
+    before do
+      @res = FactoryGirl.create(:valid_reservation,
+                                start_date: Time.zone.today,
+                                due_date: Time.zone.today + 1.day)
+    end
+    it 'should fail to edit with check in time before start date' do
+      @res.update_attributes(checked_in: Time.zone.now - 2.days)
+      expect(@res.check_in_time_after_start_date).not_to be_nil
+    end
+    it 'should fail to edit with check out time before start date' do
+      @res.update_attributes(checked_out: Time.zone.now - 2.days)
+      expect(@res.check_out_time_after_start_date).not_to be_nil
+    end
+    it 'should fail to edit with check in time before check out time' do
+      @res.update_attributes(checked_in: Time.zone.now,
+                             checked_out: Time.zone.now + 1)
+      expect(@res.check_in_time_after_check_out_time).not_to be_nil
+    end
+    it 'should not create an invalid reservation' do
```

Comment:
Aren't you only testing a single case here? (`checked_in` before `start_date`)? What about the other validations?